zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,72 @@
+use bulletproofs::r1cs;
+use bulletproofs::{BulletproofGens, PedersenGens};
+use curve25519_dalek::ristretto::CompressedRistretto;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use std::collections::VecDeque;
+
+use zkvm::*;
+
+fn issue_contract(
+    qty: u64,
+    flv: Scalar,
+    issuance_pred: &Predicate,
+    nonce_pred: &Predicate,
+) -> Vec<Instruction> {
+    vec![
+        Instruction::Push(Commitment::from(CommitmentWitness::blinded(qty)).into()), // stack: qty
+        Instruction::Var, // stack: qty-var
+        Instruction::Push(Commitment::from(CommitmentWitness::unblinded(flv)).into()), // stack: qty-var, flv
+        Instruction::Var,                                // stack: qty-var, flv-var
+        Instruction::Push(issuance_pred.clone().into()), // stack: qty-var, flv-var, pred
+        Instruction::Issue,                              // stack: issue-contract
+        Instruction::Push(nonce_pred.clone().into()),    // stack: issue-contract, pred
+        Instruction::Nonce,                              // stack: issue-contract, nonce-contract
+        Instruction::Signtx,                             // stack: issue-contract
+        Instruction::Signtx,                             // stack: issued-value
+        Instruction::Push(issuance_pred.clone().into()), // stack: issued-value, pred
+        Instruction::Output(1),                          // stack: empty
+    ]
+}
+
+#[test]
+fn issue() {
+    let (tx, txid) = {
+        let issuance_pred = match Predicate::from_witness(PredicateWitness::Key(Scalar::random(
+            &mut rand::thread_rng(),
+        ))) {
```

Comment:
instead of `match`, you could simply do `.unwrap()` to panic on Err() and unwrap the Ok() - it's not supposed to fail ever anyway.